setup/index.ts

```diff
@@ -1,11 +1,31 @@
-import { askSessionToken } from './prompt';
-import { get as getEnvValue, set as setEnvValue } from './env';
+import {
+  /* askSessionToken, */
+
+  askIOBackendHost,
+  askIOBackendBasePath
+} from "./prompt";
+import { get as getEnvValue, set as setEnvValue, EnvKey } from "../lib/env";
+import { TaskEither } from "fp-ts/lib/TaskEither";
+import { task } from "fp-ts/lib/Task";
+
+const promptIfNot = (
+  key: EnvKey,
+  promptFn: () => TaskEither<Error, string>
+) => {
+  const setter = () =>
+    promptFn().fold<string>(
+      error => { throw error },
+      value => {
+        setEnvValue(key, value);
```

Comment:
this method has a sideeffect (assign to process.env)

Reply:
I originally stored configurations in a Map on a singleton module, with env variables being default values. 
The problem I ran into is jest did not to consider the module a singleton. Maybe because i'm using the setup script as `globalSetup`. So this _workaround_ worked :)

Reply:
there should be no difference between a singleton module and a non-singleton one since modules shouldn't have side effects

Reply:
Sorry but I don't understand if the problem is that I'm changing `process.env` or it's the assignment itself

Reply:
first one. I don't get "jest did not to consider the module a singleton" (but we can go on anyway : )

Reply:
ok ok, thus resolved ;)